src/app/persistence/String.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/persistence/String.h>
+
+namespace chip::app::Storage::Internal {
+
+bool ShortString::SetContent(CharSpan value)
+{
+
+    ShortPascalString view = AsPascal();
+
+    if (!view.SetValue(value))
+    {
+        view.SetValue(""_span);
+        NullTerminate();
+        return false;
+    }
+    NullTerminate();
+    return true;
+}
+
+bool ShortStringIO::FinalizeRead(MutableByteSpan actuallyRead)
+{
+    if (!ShortPascalString::IsValid({ reinterpret_cast<const char *>(actuallyRead.data()), actuallyRead.size() }))
```

Comment:
See, if IsValid took a ByteSpan always, we would not need this casting stuff here.

Reply:
The issue here is that the span is a `CharSpan` for pascal string, where as the I/O is performed over bytes. Will add a comment.

Reply:
On 2nd thought I can just make pascal validity checks always happen over byte arrays as they should be done on I/O buffers anyway.